kategory-effects-kotlinx/src/main/kotlin/kategory/effects/data/DeferredKW.kt

```diff
@@ -0,0 +1,96 @@
+package kategory
+
+import kategory.effects.data.internal.Platform
+import kotlinx.coroutines.experimental.*
+import kotlin.coroutines.experimental.CoroutineContext
+
+typealias DeferredResult<A> = Either<Throwable, A>
+
+@higherkind
+@deriving(Monad::class, AsyncContext::class)
+data class DeferredKW<out A>(val thunk: (CoroutineContext) -> Deferred<A>) : DeferredKWKind<A> {
+
+    fun <B> map(f: (A) -> B): DeferredKW<B> =
+            flatMap { a: A -> pure(f(a)) }
+
+    fun <B> flatMap(f: (A) -> DeferredKWKind<B>): DeferredKW<B> =
+            DeferredKW { context: CoroutineContext ->
+                this@DeferredKW.attempt(context).fold(
```

Comment:
I think `attempt` defeats the purpose of having this datatype. I recall reimplementing this in an fully async way. But if we cause to block what good does deffer provides over `async/await`?